test/integration/scheduler_perf/scheduler_perf.go

```diff
@@ -816,6 +816,10 @@ type barrierOp struct {
 	// Namespaces to block on. Empty array or not specifying this field signifies
 	// that the barrier should block on all namespaces.
 	Namespaces []string
+	// Wait for pods to be attempted, so to go through at least one scheduling cycle.
+	// If false, barrier waits for pods to be scheduled.
+	// Optional
+	WaitUntilAttempted bool
```

Comment:
I really would like to make this field the opposite, like `scheduled`, but it wouldn't be backwards-compatible :(

For readability, in APIs we tend to avoid booleans https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

And here it could be handy. You can make this field be:

```yaml
  requirement: Attempted
```

And document that, when not explicitly specified, it's interpreted as `Scheduled`.

Reply:
Okay, I added a `stageRequirement` field for this.